plugins/events-backend/src/service/http/HttpPostIngressEventPublisher.ts

```diff
@@ -87,25 +98,53 @@ export class HttpPostIngressEventPublisher {
     validator?: RequestValidator,
   ): void {
     const path = `/${topic}`;
+    const logger = this.logger;
 
     router.post(path, async (request, response) => {
-      const requestDetails = {
-        body: request.body,
-        headers: request.headers,
-      };
-      const context = new RequestValidationContextImpl();
-      await validator?.(requestDetails, context);
-      if (context.wasRejected()) {
-        response
-          .status(context.rejectionDetails!.status)
-          .json(context.rejectionDetails!.payload);
-        return;
+      const requestBody = request.body;
+      if (!Buffer.isBuffer(requestBody)) {
+        throw new Error(
+          `Failed to retrieve raw body from incoming event for topic ${topic}; not a buffer: ${typeof requestBody}`,
+        );
+      }
+
+      const bodyBuffer: Buffer = requestBody;
+      const parsedContentType = contentType.parse(request);
+      const encoding = parsedContentType.parameters.charset ?? 'utf-8';
+      if (!Buffer.isEncoding(encoding)) {
+        throw new UnsupportedCharsetError(encoding);
+      }
+
+      const bodyString = bodyBuffer.toString(encoding);
+      const bodyParsed =
+        parsedContentType.type === 'application/json'
+          ? JSON.parse(bodyString)
+          : bodyString;
```

Comment:
Previously we would just throw an error if the content type wasn't supported, thinking we might as well keep doing that right? i.e. if it's not JSON

Reply:
yes, we could. Even for just now and rediscuss later. Just thought to remove this limitation that was not necessarily put in place because we must.

Reply:
I will add the rejection of other content types. We can decide later, whether we want to open this up more for other content types (and maybe only specific ones).